pinot-common/src/main/java/org/apache/pinot/common/utils/PinotDataType.java

```diff
@@ -315,6 +408,58 @@ public Double convert(Object value, PinotDataType sourceType) {
     }
   },
 
+  TIMESTAMP {
+    @Override
+    public int toInt(Object value) {
+      throw new UnsupportedOperationException("Cannot convert value from TIMESTAMP to INTEGER");
+    }
+
+    @Override
+    public long toLong(Object value) {
+      return ((Timestamp) value).getTime();
+    }
+
+    @Override
+    public float toFloat(Object value) {
+      throw new UnsupportedOperationException("Cannot convert value from TIMESTAMP to FLOAT");
+    }
+
+    @Override
+    public double toDouble(Object value) {
+      return ((Timestamp) value).getTime();
+    }
+
+    @Override
+    public boolean toBoolean(Object value) {
+      throw new UnsupportedOperationException("Cannot convert value from TIMESTAMP to BOOLEAN");
```

Comment:
I would have done zero as `false` and all other timestamp values as `true`. This convention generalizes to work with everything and type conversion becomes simple :-)

Reply:
Done

Reply:
Revert this to unsupported because I feel it makes no sense to convert a timestamp to boolean, most likely a human error of putting the wrong data type